lib/freemail.rb

```diff
@@ -28,6 +28,11 @@ def self.free_or_disposable?(email)
   end
 
   def self.get_domain(email)
-    email.split('@').last.downcase.strip
+    parts = email.split('@').map { |s| s.downcase.strip }
+    if parts.size == 2 && parts.none?(&:empty?)
+      parts.last
+    else
+      raise ArgumentError, "Invalid email: #{email}"
+    end
```

Comment:
This will split at the first '@' and return everything afterwards (including subsequent @s) and should pass the `gmail.com@` test by returning `''` and not considering that a "free" email.

```suggestion
    email.split('@')[1..-1].join('@')&.downcase&.strip
```